src/ui/overlayService/overlayService.js

```diff
@@ -0,0 +1,92 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2018, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define([
+    './overlay.vue',
+    'vue'
+], function (
+    OverlayComponent,
+    Vue
+) {
+
+    function OverlayService() {
+        this.activeOverlays = [];
+        this.overlayId = 0;
+        this.onDestroyCallbacks = {};
+    }
+
+    OverlayService.prototype.show  = function (element, onDestroyCallback) {
+        if(this.activeOverlays.length) {
+            this.activeOverlays[this.activeOverlays.length - 1].overlay.classList.add('invisible');
+        }
+
+        let overlayTypeCssClass = 'l-large-view'; // Values could be l-large-view, l-dialog, l-message
+        this.overlay = document.createElement('div');
+        this.overlay.classList.add('l-overlay-wrapper', overlayTypeCssClass);
+
+        document.body.appendChild(this.overlay);
+
+        this.component = new Vue({
```

Comment:
does this need to be a member variable of overlay service or can it just be `let component = ...`